crates/ruff_python_parser/src/typing.rs

```diff
@@ -93,7 +93,17 @@ fn parse_simple_type_annotation(
 
 fn parse_complex_type_annotation(string_expr: &ExprStringLiteral) -> AnnotationParseResult {
     let mut parsed = parse_expression(string_expr.value.to_str())?;
-    relocate_expr(parsed.expr_mut(), string_expr.range());
+    // Remove quotes from nested forward reference type annotations
+    let range_excluding_quotes =
+        if let Expr::StringLiteral(ExprStringLiteral { ref value, .. }) = *parsed.syntax.body {
+            let string_parts = value.as_slice();
+            let start = string_parts[0].flags.closer_len();
```

Comment:
Why does this use `closer_len` instead of `opener_len`?

Reply:
Ah, this was `opener_len` before but got lost somewhere in the refactoring.